File: ios/Anemone SDK/request/PaginationOptions.swift
//
//  PaginationOptions.swift
//  Anemone SDK
//
//  Created by Angel Luis Garcia on 24/03/2017.
//  Copyright © 2017 Syltek Solutions S.L. All rights reserved.
//

import Foundation

public struct PaginationOptions: Equatable {
    public var page: Int?
    public var size: Int?

    public init() { }

    public func withPage(_ page: Int?) -> PaginationOptions {
        var options = self
        options.page = page
        return options
    }

    public func withSize(_ size: Int?) -> PaginationOptions {
        var options = self
        options.size = size
        return options
    }

    var params: [String: Any] {
        var params = [String: Any]()
        if let page = page {
            params["page"] = page
        }
        if let size = size {
            params["size"] = size
        }
        return params
    }
}

File: android/app/src/main/java/com/playtomic/test/anemone_sdk/request/PaginationOptions.kt
package com.anemonesdk.service.request

/**
 * Created by agarcia on 24/03/2017.
 */

data class PaginationOptions internal constructor(
    val page: Int?,
    val size: Int?
) {

    constructor() : this(null, null)

    fun withPage(page: Int?) = copy(page = page)

    fun withSize(size: Int?) = copy(size = size)

    internal val params: Map<String, Any>
        get() {
            val params = mutableMapOf<String, Any>()
            if (page != null) {
                params["page"] = page
            }
            if (size != null) {
                params["size"] = size
            }
            return params
        }
}
